### PR TITLE
Add group by and sort by to asciidoc report generation

### PR DESCRIPTION
**Pull request by lfvjimisola**

### CODE REVIEW

**Review by lfvdavid — APPROVED** [review]
Looks good to me, quick demo done and output/structure agreed upon.

**Comment by lfvjimisola** [comment]
Ok. Thanks. Let's await approval from @lfvbarcus as well.

File: src/reqstool/commands/report/report.py [hunk 1]
```diff
@@ -122,43 +90,69 @@ def __generate_asciidoc_information(self, reqs, statistics: StatisticsContainer)
...
+        statistics_table = Jinja2Utils.render(
+            data=statistics._total_statistics, template=self.jinja2_templates[Jinja2Templates.TOTAL_STATISTICS]
+        )
+
+        grouped_requirements: Dict[str, List[UrnId]] = GroupByOrganizor(
+            cid=cid, group_by=self.group_by, sort_by=self.sort_by
+        ).grouped_requirements
+
+        # category, List(asciidoc for each req)
+        template_data: Dict[str, List[str]] = defaultdict(list)
```
Comment (lfvbarcus) [inline src/reqstool/commands/report/report.py:102]: Sonar complains about this line as it is never used. 

Perhaps it is better remove this line and to combine the type help with the declaration of the dict below instead? 


        template_data: Dict[str, List[str]] = {
            category: [self.__extract_template_data(req_template=aggregated_data[urn_id]) for urn_id in urn_ids]
            for category, urn_ids in grouped_requirements.items()
        }

File: src/reqstool/common/jinja2.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        def load_template(loader: BaseLoader) -> Template:
+            template_env = Environment(
+                loader=loader, autoescape=select_autoescape(), trim_blocks=True, lstrip_blocks=True
+            )
+            return template_env.get_template(template_name)
+
+        try:
+            template_module: Package = reqstool.commands.report.templates
+            template_path: Path = files(template_module).joinpath("")
```
Comment (lfvbarcus) [inline src/reqstool/common/jinja2.py:41]: Sonar complains about type hit to Path. Should be Traversable

File: src/reqstool/commands/report/report.py (hunk 1 above)
Comment (lfvjimisola) [reply src/reqstool/commands/report/report.py:102]: Fixed already. Pushing soon.

File: src/reqstool/common/jinja2.py (hunk 2 above)
Comment (lfvjimisola) [reply src/reqstool/common/jinja2.py]: Fixed now.

**Review by lfvbarcus — APPROVED** [review]
Nice work!

**Pull request merged**